From cbffefb483d96769fa30fbf2aa5a11cd396f16ca Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 24 Jul 2018 14:05:17 +0800
Subject: [PATCH 39/91] MMFMWK-8077 glupload: respect to downstream format
 priority

Make upstream caps format respect to downstream's priority
to try hard to choose RGBA format if possible

Upstream-Status: Pending

(cherry picked from commit 76fc25cf894e065c303cbb7684e2a3e3f8fd468e)
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst-libs/gst/gl/gstglupload.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 0eefb3a..96b554d 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -3852,7 +3852,7 @@ gst_gl_upload_transform_caps (GstGLUpload * upload, GstGLContext * context,
   }
 
   if (filter) {
-    result = gst_caps_intersect_full (filter, tmp, GST_CAPS_INTERSECT_FIRST);
+    result = gst_caps_intersect_full (tmp, filter, GST_CAPS_INTERSECT_FIRST);
     gst_caps_unref (tmp);
   } else {
     result = tmp;
-- 
2.43.0

